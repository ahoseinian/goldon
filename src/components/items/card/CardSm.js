import React from 'react'
import styled from 'styled-components'
import {Link} from 'react-router'
import ItemCard from '../../common/Card'
import ItemImage from '../Image'
import PriceRow from './PriceRow'
import Sizes from './Sizes'

const ItemName = styled.div `
  text-align:center;
  font-size: 1.2rem;
  padding-top: .7rem;
  padding-left: 1rem;
  line-height: 1.5rem;
`

const Card = ({item, addItemToCart}) => (
  <Link to={`/items/${item.id}/${item.name}`} className="text-muted">
    <ItemCard className="card">
      <div className="row">
        <div className="col-5">
          <ItemImage id={item.images[0]}/>
        </div>
        <div className="col-7">
          <ItemName>{item.name}</ItemName>
          <PriceRow price={item.price} />
          <Sizes size={item.size} />
        </div>
      </div>
    </ItemCard>
  </Link>
)

export default Card
