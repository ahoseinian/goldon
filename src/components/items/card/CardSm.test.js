import React from 'react'
import {shallow} from 'enzyme'
import CardSm from './CardSm'
import ItemImage from '../image/Image'
import toJson from 'enzyme-to-json'

const item = {
  id: '#number1',
  name: 'first item',
  images: ['one'],
  price: 22,
}

describe('CardSm', () => {
  it('renders', () => {
    const wrapper = shallow(<CardSm {...item} />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('does not renders empty image', () => {
    const item = {
      id: '1',
      name: '2',
      images: [],
      price: 22
    }
    const wrapper = shallow(<CardSm {...item} />)
    expect(wrapper.find(ItemImage).length).toEqual(0)
  })
})
