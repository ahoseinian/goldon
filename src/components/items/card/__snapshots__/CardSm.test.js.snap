exports[`CardSm renders 1`] = `
<Link
  className="text-muted"
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/items/#number1/first item">
  <styled.div
    className="card">
    <div
      className="row">
      <div
        className="col-5">
        <ItemImage
          id="one" />
      </div>
      <div
        className="col-7">
        <styled.div>
          first item
        </styled.div>
        <PriceRow
          price={22} />
      </div>
    </div>
  </styled.div>
</Link>
`;
